limbus/core/component.py

```diff
@@ -306,15 +316,15 @@ async def __call__(self) -> None:
 
     def is_stopped(self) -> bool:
         """Check if the component is stopped or is going to be stopped."""
-        if self.state in [ComponentState.STOPPED, ComponentState.STOPPED_AT_ITER,
-                          ComponentState.ERROR, ComponentState.FORCED_STOP]:
+        if self.state[0] in [ComponentState.STOPPED, ComponentState.STOPPED_AT_ITER,
+                             ComponentState.ERROR, ComponentState.FORCED_STOP]:
             return True
         return False
 
     def _stop_if_needed(self) -> bool:
         """Stop the component if it is required."""
         if self.is_stopped():
-            if self.state is not ComponentState.STOPPED_AT_ITER:
+            if self.state[0] is not ComponentState.STOPPED_AT_ITER:
```

Comment:
no guards to check random access index?

Reply:
still a draft this PR. XD A lot of work is missing. I do not like this index but it was the easy way to do it.